src/utilities/graphqlConnectionFactory.ts

```diff
@@ -1,38 +1,178 @@
-import type { ConnectionPageInfo } from "../types/generatedGraphQLTypes";
+import type {
+  ConnectionPageInfo,
+  ConnectionError,
+  CursorPaginationInput,
+} from "../types/generatedGraphQLTypes";
+import type { Types } from "mongoose";
 
 interface InterfaceConnectionEdge<T> {
   cursor: string;
   node: T;
 }
 
 interface InterfaceConnection<T> {
-  edges?: InterfaceConnectionEdge<T> | null | undefined[];
+  edges: InterfaceConnectionEdge<T>[];
   pageInfo: ConnectionPageInfo;
 }
 
+interface InterfaceConnectionResult<T> {
+  data: InterfaceConnection<T> | null;
+  errors: ConnectionError[];
+}
+
+type GetNodeFromResultFnType<T1, T2> = {
+  (result: T2): T1;
+};
+
 /*
-This is a factory function to quickly create a graphql
-connection object. The function accepts a generic type
-'T' which is used to reference the type of node that this
-connection and it's edges will reference. A node is
-a business object which can be uniquely identified in graphql.
-For example `User`, `Organization`, `Event`, `Post` etc.
-All of these objects are viable candiates for a node and
-can be paginated using graphql connections. The default object
-returned by this function represents a connection which has no
-data at all, i.e., the table/collection for that node(along with
-other constraints like filters if any) is completely empty in database.
-This object will need to be transformed according to different
-logic inside resolvers.
+This is a factory function to quickly create a graphql connection object. The function accepts a generic type
+'T' which is used to reference the type of node that this connection and it's edges will reference. A node is
+a business object which can be uniquely identified in graphql. For example `User`, `Organization`, `Event`, `Post` etc.
+All of these objects are viable candiates for a node and can be paginated using graphql connections. The default object returned by this function represents a connection which has no data at all, i.e., the table/collection for that node(along with ther constraints like filters if any) is completely empty in database. 
+This object will need to be transformed according to different logic inside resolvers.
 */
 export function graphqlConnectionFactory<T>(): InterfaceConnection<T> {
   return {
     edges: [],
     pageInfo: {
+      startCursor: null,
       endCursor: null,
       hasNextPage: false,
       hasPreviousPage: false,
-      startCursor: null,
     },
   };
 }
+// Generates the limit that can should be passed in the .limit() method
+export const getLimit = (args: CursorPaginationInput): number => {
```

Comment:
you can get away with passing just args.limit as the function argument

Reply:
I also need the cursor to check if I need to supply `limit + 1` or `limit + 2`

Reply:
my bad